### ISSUE TITLE
1.20.1 Crash when using Tree Ferilizer (from Create) on TFC Trees

### ISSUE THREAD

**Issue by TFCAliarcy** [TerraFirmaCraft/TerraFirmaCraft#2629]
Using Tree Ferilizer (from Create) on TFC Saplings will cause it to grow then crash and also crashes when attempting to load said chunks. Crash does not occur when letting TFC trees grow naturally or when using Tree Fertilizer on Vanilla saplings.

[crash-2024-02-02_14.08.50-server.txt](https://github.com/TerraFirmaCraft/TerraFirmaCraft/files/14144456/crash-2024-02-02_14.08.50-server.txt)

**Comment by alcatrazEscapee** [comment]
Looks to be a conflict where Create directly calls methods concerned with bonemeal application, but do not fire the Forge bonemeal event, which is fired in all vanilla call paths. This then calls vanilla methods for applying bone meal, which won't work for TFC trees. Should be solvable via overriding said methods (which in effect, would make tree fertilizer do nothing, which I think is acceptable).

**Comment by alcatrazEscapee** [comment]
Upon actual testing + looking into it further, I cannot reproduce this issue, nor do I see how it would actually occur. Tree fertilizer seems to work fine on every TFC sapling.

**Comment by TFCAliarcy** [comment]
I've done further testing and it seems to occur when music starts playing after using tree fertilizer, I've tested it on fresh clients both MultiMC and with Mojang's launcher and had a friend test it and they also crashed when music started playing on both fresh clients of MultiMC and Mojang.

To replicate this place down a sapling on dirt and use Create's tree fertilizer on the sapling. After having done more testing it turns out that it doesn't have to be a TFC sapling but it does require TFC to be installed. I also noticed that the green "+" effects are missing when TFC is installed, I don't know if it is a clue but it was something of note.

[crash-2024-02-05_21.42.22-server.txt](https://github.com/TerraFirmaCraft/TerraFirmaCraft/files/14174117/crash-2024-02-05_21.42.22-server.txt)

**Comment by alcatrazEscapee** [comment]
Your further testing is a _completely_ different crash and issue, namely, a duplicate of #2591. This is a crash due to corrupted partially generated chunks and has nothing to with either tree fertilizer or the onset of music playing.

**Comment by TFCAliarcy** [comment]
Here is of a fresh world, it is easier to get when rejoining a world but I was able to get this from a fresh one. I'm not saying that music starting to causes the crash, just that it coincides with it for whatever reason. It is possible to replicate this from a fresh world but easier when rejoining as when starting a new world music will often start before I can fertilize whereas rejoining a world will have it be fertilized and then once music begins to play it will crash.

I, albeit as someone with little to no coding experience, do think that the crash is due to corruption but that fertilizing the sapling is corrupting the world. I have also attached the world used in the crash report if that may aid you.

[crash-2024-02-05_22.42.17-server.txt](https://github.com/TerraFirmaCraft/TerraFirmaCraft/files/14174405/crash-2024-02-05_22.42.17-server.txt)

[New World.zip](https://github.com/TerraFirmaCraft/TerraFirmaCraft/files/14174545/New.World.zip)
